packages/dashboard/src/util/theme.ts

```diff
@@ -0,0 +1,63 @@
+import { makeStyles } from '@material-ui/core';
+import { ThemeMode } from '../settings';
+
+const darkThemeStyles = makeStyles((theme) => ({
+  components: {
+    backgroundColor: theme.palette.primary.dark,
+    color: theme.fontColors.darkTheme,
+  },
+  background: {
+    backgroundColor: theme.palette.secondary.dark,
+  },
+  map: {
+    filter:
+      'invert(90%) sepia(12%) saturate(5773%) hue-rotate(193deg) brightness(92%) contrast(92%)',
+  },
+  accordian: {
+    backgroundColor: theme.palette.secondary.dark,
+    color: theme.fontColors.darkTheme,
+  },
+  headerTab: {
+    borderRight: `0.25px solid ${theme.fontColors.darkTheme}`,
+    borderLeft: `0.25px solid ${theme.fontColors.darkTheme}`,
+  },
+  font: {
+    color: theme.fontColors.darkTheme,
+  },
+  leafletControl: {
+    color: `${theme.fontColors.darkTheme} !important`,
+    backgroundColor: `${theme.palette.primary.dark} !important`,
+  },
+}));
+
+const lightThemeStyles = makeStyles((theme) => ({
+  components: {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.fontColors.lightTheme,
+  },
+  background: {
+    backgroundColor: theme.palette.secondary.main,
+  },
+  map: {
+    // use original img, don't apply styles
+  },
+  accordian: {
+    backgroundColor: theme.palette.secondary.main,
+    color: theme.fontColors.lightTheme,
+  },
+  headerTab: {
+    borderRight: `0.25px solid ${theme.fontColors.lightTheme}`,
+    borderLeft: `0.25px solid ${theme.fontColors.lightTheme}`,
+  },
+  font: {
+    color: theme.fontColors.lightTheme,
+  },
+  leafletControl: {
+    color: `${theme.fontColors.lightTheme} !important`,
+    backgroundColor: `${theme.palette.primary.main} !important`,
+  },
+}));
+
+export const decideThemeStyle = (themeContext: ThemeMode) => {
```

Comment:
This causes a lot of prop drilling, theming is universal enough that you should use a context to pass these.

Also not sure if these are a good idea, each component should set their own styles following the theme engine. If a component is not following the theme, it should be fixed by making it use the theme properly instead of applying a fixed class on top of it.

Reply:
I guess passing in the current themeMode would be a better idea instead of passing the css, then let each component decide what style they are going to have